frontend/src/components/results/PreviewCodeContent.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Player, Submission } from '../../api/Game';
+import Language from '../../api/Language';
+import { SecondaryHeaderText } from '../core/Text';
+import ResizableMonacoEditor from '../game/Editor';
+
+const CodePreview = styled.div`
+  position: relative;
+  text-align: left;
+  margin: 10px auto;
+  
+  width: 75%;
+  height: 45vh;
+  padding: 8px 0;
+  box-sizing: border-box;
+  border: 1px solid ${({ theme }) => theme.colors.blue};
+  border-radius: 8px;
+`;
+
+type PreviewCodeContentProps = {
+  players: Player[],
+  playerIndex: number,
+}
+
+function PreviewCodeContent(props: PreviewCodeContentProps) {
+  const { players, playerIndex } = props;
+
+  if (!players[playerIndex] || !players[playerIndex].submissions.length) {
+    return null;
+  }
+
+  let bestSubmission: Submission | undefined;
+  players[playerIndex].submissions.forEach((submission) => {
+    if (!bestSubmission || submission.numCorrect > bestSubmission.numCorrect) {
+      bestSubmission = submission;
+    }
+  });
```

Comment:
Sounds good, and no problem at all 👍

Reply:
Done! ✅

Reply:
Actually, upon implementing this, it did not work as predicted; for whatever reason, the code would lag "one" behind, showing the results of the last player looked at rather than the player themselves.

Do you have any idea why this might be @alankbi? I thought maybe it's because the editor may not refresh on default value change, and since the best submission is updated via `useEffect`, it does not update alongside that.

I left the new code in right now, but it's not working correctly, so I'll want to fix that before merge - lmk if you have any ideas on this.

Reply:
Hm that's interesting, yeah it could be what you were suggesting - not sure tbh what a good solution would be, but I'm totally fine reverting it, whichever is easiest

Reply:
Ok, I'll revert this back - I may switch this button to live code viewing in the next PR anyway, so we'll see but it may be a moot point.

Reply:
Reverted back - I realized it's still used on the Results page so it will still matter 👍